api/v1alpha1/texttemplate_types.go

```diff
@@ -20,16 +20,21 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	TextTemplateFinalizer = "finalizers.templates.kluctl.io"
+)
+
 // TextTemplateSpec defines the desired state of TextTemplate
 type TextTemplateSpec struct {
 	// Suspend can be used to suspend the reconciliation of this object.
 	// +optional
 	// +kubebuilder:default:=false
 	Suspend bool `json:"suspend"`
 
-	// The name of the Kubernetes service account to impersonate
-	// when reconciling this TextTemplate. If omitted, the "default" service account is used.
+	// ServiceAccountName specifies the name of the Kubernetes service account to impersonate
+	// when reconciling this TextTemplate. If omitted, the "default" service account is used
 	// +optional
+	// +kubebuilder:default:="default"
 	ServiceAccountName string `json:"serviceAccountName,omitempty"`
```

Comment:
```suggestion
	ServiceAccountName *string `json:"serviceAccountName,omitempty"`
```

Using a pointer here will make it possible to distinguish `""` (invalid) from unset. Optional fields should usually be represented with pointer in the Go API, ref. https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
I wonder if this is really an optional field when defaulting is also in place. In the end, the field will always have a non-nil value. What do you think?

Reply:
FYI, I'll merge for now and then include this change in a new PR after it's clear what to do here.